src/actions/ExchangeRateActions.js

```diff
@@ -17,79 +32,88 @@ export const updateExchangeRates = () => async (dispatch: Dispatch, getState: Ge
-    exchangeRates[`${currencyCode}_${accountIsoFiat}`] = rateCache.convertCurrency(currencyCode, accountIsoFiat)
-    exchangeRates[`${currencyCode}_iso:USD_${yesterdayDate}`] = fetchExchangeRateHistory(currencyCode, yesterdayDate)
-    // add them to the list of promises to resolve
-    // keep track of the exchange rates
+    exchangeRates.push({ currency_pair: `${currencyCode}_${walletIsoFiat}` })
+    exchangeRates.push({ currency_pair: `${currencyCode}_${accountIsoFiat}` })
+    exchangeRates.push({ currency_pair: `${currencyCode}_iso:USD`, date: `${yesterdayDate}` })
     // now add tokens, if they exist
     if (walletIsoFiat !== 'iso:USD') {
-      exchangeRates[`iso:USD_${walletIsoFiat}`] = rateCache.convertCurrency('iso:USD', walletIsoFiat)
+      exchangeRates.push({ currency_pair: `iso:USD_${walletIsoFiat}` })
     }
     for (const tokenCode of Object.keys(wallet.balances)) {
       if (tokenCode !== currencyCode) {
-        exchangeRates[`${tokenCode}_${walletIsoFiat}`] = rateCache.convertCurrency(tokenCode, walletIsoFiat)
-        exchangeRates[`${tokenCode}_${accountIsoFiat}`] = rateCache.convertCurrency(tokenCode, accountIsoFiat)
-        exchangeRates[`${tokenCode}_iso:USD_${yesterdayDate}`] = fetchExchangeRateHistory(tokenCode, yesterdayDate)
+        exchangeRates.push({ currency_pair: `${tokenCode}_${walletIsoFiat}` })
+        exchangeRates.push({ currency_pair: `${tokenCode}_${accountIsoFiat}` })
+        exchangeRates.push({ currency_pair: `${tokenCode}_iso:USD`, date: `${yesterdayDate}` })
       }
     }
   }
-  const exchangeRateKeys = Object.keys(exchangeRates)
-  const exchangeRatePromises = Object.values(exchangeRates)
-  // Promise.allSettled() is the correct function for this but somehow not included in Promise
-  const rates = await Promise.all(
-    exchangeRatePromises.map(promise =>
-      // $FlowExpectedError - Object.values() always produce mixed type so .catch will produce error
-      promise.catch(e => {
-        console.log(e)
-        return 0
-      })
-    )
-  )
-
-  for (let i = 0; i < exchangeRateKeys.length; i++) {
-    const key = exchangeRateKeys[i]
-    const codes = key.split('_')
-    const reverseExchangeRateKey = `${codes[1]}_${codes[0]}${codes[2] ? '_' + codes[2] : ''}`
-    const rate = rates[i]
-    finalExchangeRates[key] = '0'
-    finalExchangeRates[reverseExchangeRateKey] = '0'
 
-    if (rate != null && !isNaN(rate) && rate !== 0) {
-      const rateStr = rate.toFixed(DECIMAL_PRECISION)
-      finalExchangeRates[key] = rateStr
-      finalExchangeRates[reverseExchangeRateKey] = div('1', rateStr, DECIMAL_PRECISION)
+  const serverRates: GuiExchangeRates = {}
+  for (let startIndex = 0; startIndex < exchangeRates.length; startIndex += RATES_SERVER_MAX_QUERY_SIZE) {
+    let endIndex = startIndex + RATES_SERVER_MAX_QUERY_SIZE - 1
+    if (endIndex > exchangeRates.length) {
+      endIndex = exchangeRates.length
     }
+    const query = exchangeRates.slice(startIndex, endIndex)
```

Comment:
This works but I think it can be a bit easier to read as:
```js
let exchangeRates = []

while (exchangeRates.length > 0) {
    const query = exchangeRates.splice(0, RATES_SERVER_MAX_QUERY_SIZE)
    ....
}
```

Reply:
sounds good.